bitcoin-magazine/bitcoin-explained/explaining-rgb-tokens-the-technical-side-of-bitcoin.md

```diff
@@ -0,0 +1,1254 @@
+And that was all to get to RGB.
+
+Sjors Provoost: 00:20:31
+
+That's right.
+
+## What Is RGB?
+
+Aaron van Wirdum: 00:20:32
+
+Sjors, what is RGB?
+
+Sjors Provoost: 00:20:34
+
+Oh, do I have to explain it?
+Ruben was gonna explain it.
+
+Ruben Somsen: 00:20:37
+
+Okay, so with RGB, it's a very similar system in the sense that we still have these tokens that are being generated and they're being moved on the Bitcoin blockchain.
+But what we're doing here is we're using the existing UTXOs as a kind vessel into which we add and move the coins there.
+So let's say I have a UTXO and I have some RGB tokens in there.
+Well, first I would have to generate them, but let's skip that for a second.
+So there's an output, it's mine.
+There are 10 USD Tether RGB tokens in there.
+
+Aaron van Wirdum: 00:21:14
+
+When you say in there, what do you mean?
+
+Ruben Somsen: 00:21:17
+
+Somehow, and that is of course like, so there needs to be some kind of genesis moment where the tokens were generated.
+So the whole point of this system is that it allows you to create tokens and then spend them.
+And so first, the USD Tether people would have to issue this token.
+And generally, the way that's done in RGB is actually you do use an OP_RETURN.
+So first, there's going to be a transaction.
+The transaction has an OP_RETURN, so everybody can see it.
+And that transaction says, hereby, I declare that now inside of this new UTXO there are just 1 million dollars worth of USD Tether.
+
+Sjors Provoost: 00:21:52
+
+Yeah, and this is actually where your hash at least comes up, because what is in the OP_RETURN as far as I know is a hash of a JSON file.
+So anybody who has the JSON file will know that the hash refers to it.
+And they can then read the JSON file and see what the supply is and what the rules are for whatever asset has been created.
+
+Aaron van Wirdum: 00:22:14
+
+A JSON file is just text, right?
+
+Sjors Provoost: 00:22:17
+
+It's just text.
+Yeah.
+With annoying brackets.
+
+Ruben Somsen: 00:22:20
+
+So, okay, so now we have a UTXO with $1 million in their USD Tether dollars IOUs.
+And when they first send them to a buyer, somebody who says, "Okay, I'll give you $1,000, you give me $1,000 worth of these coins."
+So the first thing that...
+
+## Client-Side Validation
+
+Ruben Somsen: 00:22:37
+
+...RGB does, is client-side validation, meaning that the JSON file that Sjors just pointed out needs to be transferred during the coin transfer.
+So as soon as these coins move from one owner to the next owner, you also have to provide all the data, which refers to the hash that you put in the Bitcoin blockchain.
+
+Aaron van Wirdum: 00:22:55
+
+Right.
+
+Sjors Provoost: 00:22:56
+
+So the first person who buys the tether, basically, has to receive an email.
+And inside that email should be the JSON file, and I guess also a proof that they now own the coins.
+
+Aaron van Wirdum: 00:23:09
+
+Right.
```

Comment:
"Yeah", "Sure" or "Right" are often used as filler words at the beginning of a speaker's segment. Please remove these occurrences for clarity and accuracy.

Reply:
Sometimes, Yes and Right as answers that a speaker is giving to the previous speaker's question. But yeah, in this particular case, it does seem unnecessary.

Reply:
Definitely! That's why I used the word "often", because as you said sometimes the usage of those words does make sense.